### ISSUE TITLE
No source context even though source bundles available

### ISSUE THREAD

**Issue by bruno-garcia** [getsentry/sentry#60236]
![image](https://github.com/getsentry/sentry/assets/1633368/1af0eb90-10f0-4c93-ac55-f1caac8de7ae)

Event in Sentry: https://bruno-garcia.sentry.io/issues/4647134873/?imageCodeId=65594f7b12000&imageDebugId=b951abf3-1d87-42a9-a886-65880caa884c-e185dcc0&project=4506249065332736&query=is%3Aunresolved&referrer=issue-stream&statsPeriod=14d&stream_index=0

![image](https://github.com/getsentry/sentry/assets/1633368/f4c2bd7a-d4e6-4ffa-8154-fb7c5e0620bf)

PDB (renamed to `txt` so GH lets me upload it):
[5cd89838-5d43-4156-ae7d-cdceb9c721a0-eee65b5b.pdb.txt](https://github.com/getsentry/sentry/files/13401772/5cd89838-5d43-4156-ae7d-cdceb9c721a0-eee65b5b.pdb.txt)

DLL (renamed to `txt` so GH lets me upload it):
[Sonarr.dll.txt](https://github.com/getsentry/sentry/files/13401771/Sonarr.dll.txt)

**Comment by bruno-garcia** [comment]
@vaind since you looked at this, the fact that works on bundled code in the PDB but not in source bundles, any hints where the problem might be or where to look?

**Comment by zhuxb711** [comment]
I also met this issue. C# source bundles available through setting "SentryUploadSources" on MSBuild but in the end no source available for the stack frame.

**Comment by vaind** [comment]
This has slipped through the cracks. I'll check what's going on. At the first glance, I'd say the debug symbols (PDBs) uploaded already contain _some_ source so the source bundle is not used at all.

**Comment by zhuxb711** [comment]
@vaind Yes I think so. Sentry said that the source is bundled with pdb file and have 'source' mark on it. But somehow did not used at all.

**Comment by vaind** [comment]
> [<img alt="" width="16" height="16" src="https://avatars.githubusercontent.com/u/6349682?s=64&amp;u=6e3636d7ce3521a5d9ffe6fe96b0a3ad82a2545c&amp;v=4">@vaind](https://github.com/vaind) Yes I think so. Sentry said that the source is bundled with pdb file and have 'source' mark on it. But somehow did not used at all.

If I remember correctly, PDBs often have source code blocks for generated code, even though you don't enable `EmbedAllSources` in your csproj. And the [symbolicator](https://github.com/getsentry/symbolicator) will only consider the first debug file that has sources and won't look for another one.

This is the relevant code in the Symbolicator:

https://github.com/getsentry/symbolicator/blob/2474a74225d1b8488d909e0441c1996c0078b6af/crates/symbolicator-native/src/symbolication/module_lookup.rs#L312-L333

Notably from `objects.find()` docs: "    /// Fetches matching objects and returns the metadata of the most suitable object." - it returns a single object.

And I've just found out this issue is a duplicate of https://github.com/getsentry/symbolicator/issues/1112

**Comment by zhuxb711** [comment]
Hope this issue would be able to resolve soon~ Thanks @vaind

**Comment by olksdr** [comment]
Reopen if the problem occurs  again.

**Comment by Dav1dde** [comment]
Issue was also reported as [https://github.com/getsentry/symbolic/issues/905](https://github.com/getsentry/symbolic/issues/905) and fixed, please report back if it's still an issue.

**Comment by zhuxb711** [comment]
@Dav1dde @olksdr Which Sentry version include the fixing? Need to update the .Net Sentry on App side?

**Comment by Dav1dde** [comment]
Should be in the 25.5 release, possibly 25.4. You'll just have to update Sentry, not the app.<br><br>If it doesn't work, you may have had a different issue, then please open a new issue.

**Comment by zhuxb711** [comment]
@Dav1dde I'm still unable to show the source mapping for my .Net project, even the source is included. I'm using managed sentry instead of self-host sentry.

![Image](https://github.com/user-attachments/assets/3ba5632b-168e-4070-a82b-f72725670b27)

![Image](https://github.com/user-attachments/assets/eb7f4a3b-30d0-4523-87af-3002ad6424a8)
